### PR TITLE
feat: add colored cable like items

### LINKED ISSUE #320: Add colored variants for cable-like items
Importer, exporter, external storage, etc.

Needs #319 

Make sure to respect #323

### PR DESCRIPTION
**Pull request by bulldog98**
closes: #320

### CODE REVIEW

File: refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/content/Items.java [hunk 1]
```diff
@@ -161,14 +162,22 @@ public List<Supplier<ControllerBlockItem>> getRegularControllers() {
...
+    public void addCable(final Supplier<NamedBlockItem> supplier) {
         allCables.add(supplier);
     }
 
-    public List<Supplier<CableBlockItem>> getCables() {
+    public List<Supplier<NamedBlockItem>> getCables() {
         return Collections.unmodifiableList(allCables);
     }
 
+    public void addExporters(final Supplier<NamedBlockItem> supplier) {
```
Comment (raoulvdberge) [inline refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/content/Items.java]:
```suggestion
    public void addExporter(final Supplier<NamedBlockItem> supplier) {
```

File: refinedstorage2-platform-common/src/main/resources/assets/refinedstorage2/models/item/exporter/base.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "parent": "block/block",
```
Comment (raoulvdberge) [inline refinedstorage2-platform-common/src/main/resources/assets/refinedstorage2/models/item/exporter/base.json:2]: There's no need to put `exporter.json` under a "base" folder. After all, this model file isn't used as a "base" anywhere for other model files. It's just included in the multipart definition.

File: refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/item/block/CableBlockItem.java [hunk 3]
```diff
@@ -3,7 +3,7 @@
 import net.minecraft.network.chat.Component;
 import net.minecraft.world.level.block.Block;
 
-public class CableBlockItem extends AbstractNamedBlockItem {
+public class CableBlockItem extends NamedBlockItem {
```
Comment (raoulvdberge) [inline refinedstorage2-platform-common/src/main/java/com/refinedmods/refinedstorage2/platform/common/item/block/CableBlockItem.java:6]: `CableBlockItem` can be removed I think?

File: refinedstorage2-platform-fabric/src/main/java/com/refinedmods/refinedstorage2/platform/fabric/ClientModInitializerImpl.java [hunk 4]
```diff
@@ -71,7 +71,9 @@ public void onInitializeClient() {
 
     private void setRenderLayers() {
         BlockRenderLayerMap.INSTANCE.putBlock(Blocks.INSTANCE.getImporter(), RenderType.cutout());
-        BlockRenderLayerMap.INSTANCE.putBlock(Blocks.INSTANCE.getExporter(), RenderType.cutout());
+        Blocks.INSTANCE.getExporter().values().forEach(block ->
+            BlockRenderLayerMap.INSTANCE.putBlock(block, RenderType.cutout())
+        );
```
Comment (raoulvdberge) [inline refinedstorage2-platform-fabric/src/main/java/com/refinedmods/refinedstorage2/platform/fabric/ClientModInitializerImpl.java]:
```suggestion
        Blocks.INSTANCE.getExporter().values().forEach(
            block -> BlockRenderLayerMap.INSTANCE.putBlock(block, RenderType.cutout())
        );
```

**Review by raoulvdberge — CHANGES_REQUESTED** [review]

**Comment by raoulvdberge** [comment]
To fix the ordering issues in the datagen I think that `PROPERTY_BY_DIRECTION` needs to be ordered.

File: refinedstorage2-platform-common/src/main/resources/assets/refinedstorage2/models/item/exporter/base.json (hunk 2 above)
Comment (bulldog98) [reply refinedstorage2-platform-common/src/main/resources/assets/refinedstorage2/models/item/exporter/base.json:2]: Under what name and folder do you suggest to put it? Also that was to keep it in the same vain as the cable/base.json for items.

**Comment by bulldog98** [comment]
Do we want the window that is opened on exporter click to also say e.g Cyan Exporter instead of Exporter?

**Comment by bulldog98** [comment]
> To fix the ordering issues in the datagen I think that `PROPERTY_BY_DIRECTION` needs to be ordered.

Yes I had a read at `Map.of` and the implementing `ImmutableMap`  uses a random access order, that is only garantieed to be stable inside one jvm lifetime.

File: refinedstorage2-platform-common/src/main/resources/assets/refinedstorage2/models/item/exporter/base.json (hunk 2 above)
Comment (raoulvdberge) [reply refinedstorage2-platform-common/src/main/resources/assets/refinedstorage2/models/item/exporter/base.json:2]: Nvm, this is fine!

**Review by raoulvdberge — APPROVED** [review]

**Pull request merged**